### PR TITLE
[MER-1206] Display only two decimal points in grading scores

### PR DESCRIPTION
**Pull request by julmontesdeoca**
[MER-1206](https://eliterate.atlassian.net/browse/MER-1206)

### What does it change?

- Formats score for limiting to two decimal points in the displaying of scores in the following pages:
    - Student progress view
    - Student resource progress view
    - Section's Gradebook view
    - Attempt review (student and instructor)

When editing a resoure's score manually, a number with N decimals numbers is accepted, but following the same logic as everywhere else, the number get's rounded up and limited to two significant decimals for display.

### CODE REVIEW

File: test/oli_web/live/student_view_live_test.exs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      student: student
+    } do
+      redirect_path = "/sections/#{section.slug}/enroll"
+
+      {:error, {:redirect, %{to: ^redirect_path}}} =
+        live(conn, live_view_student_view_route(section.slug, student.id))
+    end
+  end
+
+  describe "student cannot access when is logged in as an author but is not a system admin" do
```
Comment (mcass19) [inline test/oli_web/live/student_view_live_test.exs:34]: Typo: `user` or `author` instead of `student`?

File: lib/oli_web/live/common/utils.ex [hunk 2]
```diff
@@ -35,6 +35,32 @@ defmodule OliWeb.Common.Utils do
...
+    iex> format_score(78.479)
+    78.48
+
+    iex> format_score(0.0)
+    0.0
+  """
+  @spec format_score(float) :: float
+  def format_score(score) when is_float(score) do
+    formatted_score = Float.round(score, 2)
+    if score - formatted_score != 0.0, do: formatted_score, else: score
```
Comment (gastonabella) [inline lib/oli_web/live/common/utils.ex:61]: I might be missing something, but is this line necessary? If `score - formatted_score` is `0` it means they are equal, so you can just return always the `formatted_score`

File: lib/oli_web/live/progress/student_resource_view.ex [hunk 3]
```diff
@@ -57,8 +57,14 @@ defmodule OliWeb.Progress.StudentResourceView do
...
               case resource_access do
-                nil -> nil
-                _ -> ResourceAccess.changeset(resource_access, %{})
+                nil ->
+                  nil
+
+                _ ->
+                  ResourceAccess.changeset(resource_access, %{
+                    # limit score decimals to two significant figures, rounding up
+                    score: Float.round(resource_access.score, 2)
```
Comment (gastonabella) [inline lib/oli_web/live/progress/student_resource_view.ex:66]: Do we actually want to round the score in the database as well? I thought we wanted to store it as is, and then format the value only when it's displayed.

**Review by gastonabella — COMMENTED** [review]
Great job with the tests 👏

File: lib/oli_web/live/progress/student_resource_view.ex (hunk 3 above)
[the remark this answers is not in the document]
Comment (julmontesdeoca) [reply lib/oli_web/live/progress/student_resource_view.ex:66]: There's not anymore, updating.

**Review by darrensiegel — APPROVED** [review]

**Pull request merged**